org-ql.el

```diff
@@ -448,6 +452,29 @@ Returns cons (INHERITED-TAGS . LOCAL-TAGS)."
                  org-ql-tags-cache))
       (puthash position all-tags tags-cache))))
 
+(defun org-ql--expand-tag-groups (tags &optional exclude-groups)
+  (if (eq tags 'org-ql-nil)
+      'org-ql-nil
+    (let ((exclude-groups (append tags exclude-groups))
+          result)
+      (let (group-tags)
+        (dolist (tag tags)
+          (dolist (groups org-tag-groups-alist)
+            (when (and (not (cl-member (car groups) exclude-groups
+                                       :test 'string=))
+                       (cl-some (lambda (x)
+                                  (if (string-match-p "^[{].+[}]$" x)
```

Comment:
Nitpick: This `if` form doesn't seem to be indented correctly (note that Elisp indents the THEN form more than the ELSE form).  I recommend using `aggressive-indent-mode` for writing Elisp, so code is always indented correctly.

Extra nitpick: I have a strong aversion to one-symbol-per-line code, much more than my aversion to lines longer than a certain length.  In this case, especially, spreading the second `string-match-p` form across 3 lines makes it difficult to see the shape of the `if` form.

Reply:
I am also using `aggressive-indent-mode` so I am not sure what happened here. Nevertheless, I put everything in one line, so it should be clearer now. Addressed in the new commit.